core/src/surplus.ts

```diff
@@ -0,0 +1,52 @@
+import type { SurplusAuctionData, SurplusAuctionStates } from './types';
+import { getEarliestDate } from './helpers/getEarliestDate';
+import BigNumber from './bignumber';
+import getContract from './contracts';
+
+export const fetchSurplusAuctionByIndex = async function (
+    network: string,
+    auctionIndex: number
+): Promise<SurplusAuctionData> {
+    const contract = await getContract(network, 'MCD_FLAP');
+
+    const auctionsQuantityBinary = await contract.kicks();
+    const auctionsQuantity = new BigNumber(auctionsQuantityBinary._hex);
+    if (auctionsQuantity.lt(auctionIndex)) {
+        throw new Error('No active auction found with this id');
+    }
```

Comment:
I don't think you need to do extra call to check again that index doesn't exceed `kicks`. You can at least move few lines down, after you already know that there is no auction with this id and you need to check if auction never existed. 

I would also move `auctionLastIndex` logic into a separate function, since we will need to reuse it later on